src/picongpu/include/particles/manipulators/ProtonTimesWeightingImpl.hpp

```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright 2015 Axel Huebl
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+
+#pragma once
+
+#include "particles/manipulators/ProtonTimesWeightingImpl.def"
+#include "particles/traits/GetAtomicNumbers.hpp"
+
+#include "simulation_defines.hpp"
+
+namespace picongpu
+{
+namespace particles
+{
+namespace manipulators
+{
+
+struct ProtonTimesWeightingImpl
+{
+
+    template<typename T_SpeciesType>
+    struct apply
+    {
+        typedef ProtonTimesWeightingImpl type;
+    };
+
+    HINLINE ProtonTimesWeightingImpl(uint32_t currentStep)
+    {
+
+    }
+
+    template<typename T_Particle1, typename T_Particle2>
```

Comment:
The naming confused me because I thought you use `Particle1` to manipulate `Particle2`, but it is vice versa. 
In order to avoid such  confusions could you rename the types and variables to something like:
```
Particle1 -> ParticleElectron or ParticleToSet
Particle2 -> ParticleIon or ParticleWithProtons
```